### PR TITLE
OpenPaaS-Suite/esn-frontend-admin#67 - Move inboxEmailer Directives To esn.inbox.libs

### PR DESCRIPTION
**Pull request by piavgh**
Partially resolves OpenPaaS-Suite/esn-frontend-admin#67

### CODE REVIEW

File: src/esn.inbox.libs/app/directives/inbox-emailer-display/inbox-emailer-display.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const _ = require('lodash');
+
+angular.module('esn.inbox.libs')
+
```
Comment (tuanlc) [inline src/esn.inbox.libs/app/directives/inbox-emailer-display/inbox-emailer-display.js:6]:
```suggestion
```

File: src/esn.inbox.libs/app/directives/inbox-emailer-display/inbox-emailer-display.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  .directive('inboxEmailerDisplay', function(emailSendingService) {
+    function link(scope) {
+      var groupLabels = { to: 'To', cc: 'CC', bcc: 'BCC' },
+        groups = _.keys(groupLabels);
```
Comment (tuanlc) [inline src/esn.inbox.libs/app/directives/inbox-emailer-display/inbox-emailer-display.js:10]:
```suggestion
      const groupLabels = { to: 'To', cc: 'CC', bcc: 'BCC' };
      const groups = Object.keys(groupLabels);
```

File: src/esn.inbox.libs/app/directives/inbox-emailer-display/inbox-emailer-display.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+      _init();
+
+      function findAndAssignPreviewEmailer(find) {
+        for (var i = 0; i < groups.length; i++) {
+          var group = groups[i],
+            emailer = find(scope.email[group]);
```
Comment (tuanlc) [inline src/esn.inbox.libs/app/directives/inbox-emailer-display/inbox-emailer-display.js:17]:
```suggestion
          const group = groups[i];
          const emailer = find(scope.email[group]);
```

File: src/esn.inbox.libs/app/directives/inbox-emailer-display/inbox-emailer-display.js (hunk 1 above)
Comment (piavgh) [reply src/esn.inbox.libs/app/directives/inbox-emailer-display/inbox-emailer-display.js:6]: I think this is not necessary, we have many places in our code that have this convention. A blank line makes it easier for eyes

File: src/esn.inbox.libs/app/directives/inbox-emailer-display/inbox-emailer-display.spec.js [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    }
+
+    $compile(element)($scope);
+    $scope.$digest();
+
+    return element;
+  }
+
+  beforeEach(function() {
+    angular.mock.module('linagora.esn.unifiedinbox');
```
Comment (huy-ta) [inline src/esn.inbox.libs/app/directives/inbox-emailer-display/inbox-emailer-display.spec.js:27]:
```suggestion
    angular.mock.module('esn.inbox.libs');
```

**Review by chamerling — APPROVED** [review]

**Review by rezk2ll — APPROVED** [review]

File: src/esn.inbox.libs/app/controllers/resolve-emailer.controller.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+    self.resolveAvatar = resolveAvatar;
+
+    $scope.$watch('emailer', function(emailer) {
+      if (emailer && emailer.resolve) {
+        emailer.resolve();
+      }
+    });
+
+    /////
```
Comment (billyfrost418) [inline src/esn.inbox.libs/app/controllers/resolve-emailer.controller.js:16]: redundant slashes ???

**Comment by huy-ta** [comment]
Why is this "Partially resolves"?

File: src/esn.inbox.libs/app/controllers/resolve-emailer.controller.js (hunk 5 above)
Comment (piavgh) [reply src/esn.inbox.libs/app/controllers/resolve-emailer.controller.js:16]: No, it's a convention in our old code to separate sections. So I will just keep them

**Comment by piavgh** [comment]
@huy-ta: because there is also a PR from inbox-james to resolve the issue

**Review by huy-ta — APPROVED** [review]

**Pull request merged**